python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx

```diff
@@ -0,0 +1,216 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: language_level=3
+
+import numpy as np
+
+cimport cuvs.common.cydlpack
+
+from cuvs.common.resources import auto_sync_resources
+
+from cython.operator cimport dereference as deref
+from libc.stdint cimport uint32_t
+from libcpp cimport bool
+
+from cuvs.common cimport cydlpack
+from cuvs.distance_type cimport DistanceType
+
+from pylibraft.common import auto_convert_output, cai_wrapper, device_ndarray
+from pylibraft.common.cai_wrapper import wrap_array
+from pylibraft.common.interruptible import cuda_interruptible
+from pylibraft.distance.pairwise_distance import DISTANCE_TYPES
+from pylibraft.neighbors.common import _check_input_array
+
+from cuvs.common.c_api cimport cuvsResources_t
+
+from cuvs.common.exceptions import check_cuvs
+
+
+cdef class FlatIndex:
+    """
+    Brute Force index object. This object stores the trained Brute Force
+    which can be used to perform nearest neighbors searches.
+    """
+
+    cdef cuvsBruteForceIndex_t index
+    cdef bool trained
+
+    def __cinit__(self):
+        self.trained = False
+        check_cuvs(cuvsBruteForceIndexCreate(&self.index))
+
+    def __dealloc__(self):
+        if self.index is not NULL:
+            check_cuvs(cuvsBruteForceIndexDestroy(self.index))
+
+    @property
+    def trained(self):
+        return self.trained
+
+    def __repr__(self):
+        return "FlatIndex(type=BruteForce)"
+
+
+@auto_sync_resources
+def build(dataset, metric="sqeuclidean", metric_arg=2.0, resources=None):
+    """
+    Build the Brute Force index from the dataset for efficient search.
+    The following distance metrics are supported:
+        - L2
+
+    Parameters
+    ----------
+    dataset : CUDA array interface compliant matrix shape (n_samples, dim)
+        Supported dtype [float, int8, uint8]
+    metric : Distance metric to use. Default is sqeuclidean
+    metric_arg : value of 'p' for Minkowski distances
+    {resources_docstring}
+
+    Returns
+    -------
+    index: cuvs.neighbors.brute_force.FlatIndex
+
+    Examples
+    --------
+
+    >>> import cupy as cp
+    >>> from cuvs.neighbors import brute_force
+    >>> n_samples = 50000
+    >>> n_features = 50
+    >>> n_queries = 1000
+    >>> k = 10
+    >>> dataset = cp.random.random_sample((n_samples, n_features),
+    ...                                   dtype=cp.float32)
+    >>> index = brute_force.build(dataset, metric="cosine")
+    >>> distances, neighbors = brute_force.search(index, dataset, k)
+    >>> distances = cp.asarray(distances)
+    >>> neighbors = cp.asarray(neighbors)
```

Comment:
why do we need to cp.asarray these? Also, for the example we could consider printing them (same comment applies to all examples)